### PR TITLE
Un-deprecate timeout arg in pipelines

### PR DESCRIPTION
**Pull request by Rocketknight1**
As part of the pipeline sync project, this PR allows some arguments in `transformers` that don't have to be in the Hub spec. This is limited to arguments that do not actually affect the core pipeline, like `timeout`, which controls the timeout for downloading images passed as path arguments.

### CODE REVIEW

**Comment by Rocketknight1** [comment]
cc @LysandreJik @Wauplin for review!

File: tests/test_pipeline_mixin.py
Comment (Wauplin) [inline tests/test_pipeline_mixin.py:919]: Maybe over-engineering things here but I would make this list pipeline-specific

Comment (Wauplin) [reply tests/test_pipeline_mixin.py:919]: (feel free to ignore though, we can also do it next time a "transformers"-only arg is added)

**Review by Wauplin — APPROVED** [review]
Thanks @Rocketknight1! Looks good. Let's wait for another review though

File: tests/test_pipeline_mixin.py
Comment (Rocketknight1) [reply tests/test_pipeline_mixin.py:919]: Yeah, it shouldn't be too painful to make that change when we need to!

**Review by LysandreJik — APPROVED** [review]
Yes! Thanks @Rocketknight1

**Pull request merged**
